sdk/storage/src/core/authorization_policy.rs

```diff
@@ -0,0 +1,256 @@
+use crate::clients::{ServiceType, StorageCredentials};
```

Comment:
I assume most of this is being worked on in #568, right? It makes it hard to review when the changes are mixed together.

Reply:
Unfortunately I only saw the PR after submitting this. WHile you are absolutely right, my main challenge was that I needed to be able to use master key auth to get the tests to work so I found no straight forward way to separate this when migrating the core crate. But we definitely need to align how that feature gets merged. (@thovoll)